apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoader.java

```diff
@@ -436,8 +437,9 @@ private List<RifRecordLoadResult> process(
           Object recordInDb = entityManager.find(record.getClass(), recordId);
           timerIdempotencyQuery.close();
 
-          // Log if we have a non-2022 enrollment year INSERT
-          if (isBackdatedBene(rifRecordEvent)) {
+          // Log if we have a non-2022 enrollment year INSERT and we are
+          // NOT processing synthetic data.
+          if (!isSyntheticData && isBackdatedBene(rifRecordEvent)) {
```

Comment:
Since this condition it getting more complex and appears three times we may want to factor it out into a new well named method that encapsulates the concepts

```java
private boolean shouldBeSkipped(RifRecordEvent event) {
   // Don't skip any events when processing synthetic data
   if (event.getFileEvent().getParentFilesEvent().isSyntheticData())
      return false;
   else
      return isBackdatedBene(rifRecordEvent);
}
```

Reply:
sort of; I do not want to exec _event.getFileEvent().getParentFilesEvent().isSyntheticData()_ as noted in your example. The processing is in a for loop, which would result is something being fetched (i.e., determination of synthetic data) hundreds and hundreds and hundreds of thousands of times, when we only need it once (cuz it'll be for the who parentFilesEnvent).

Reply:
ok, thats fine although I would be surprised if the java compiler wasn't smart enough to inline that function and figure that out for us.